### PR TITLE
[Ide] Improve timing representation

### PR DESCRIPTION
**Pull request by rodrmoya**
Instead of hiding times that are < 1 second, always display them, as if not,
it really confuses users.

Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/571671

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputTreeCellView.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ void DrawNodeInformation (Context ctx, Xwt.Rectangle cellArea)
 			if (!buildOutputNode.HasChildren)
 				return;
 
-			var duration = buildOutputNode.GetDurationAsString ();
+			var duration = buildOutputNode.GetDurationAsString (buildOutputNode as FilteredBuildOutputNode == null);
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputTreeCellView.cs]: buildOutputNode.GetDurationAsString (buildOutputNode is FilteredBuildOutputNode) will work

Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputTreeCellView.cs]: not really, if it is a `FilteredBuildOutputNode`, we're not displaying diagnostics, and we want false to be sent in that case. But anyway, removed this "hack" in the next commit

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs [hunk 2]
```diff
@@ -506,6 +506,14 @@ void ProcessLogs (bool showDiagnostics)
 
 		public bool IsSearchInProgress => currentSearch != null && currentSearch.MatchesCount > 0;
 
+		#region IBuildOutputContextProvider
+
+		public bool IsShowingDiagnostics => showDiagnosticsButton.Active;
+
+		public string SearchString => searchEntry.Entry.Text;
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputWidget.cs]: this is not used

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/IBuildOutputContextProvider.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// THE SOFTWARE.
+
+using System;
+
+namespace MonoDevelop.Ide.BuildOutputView
+{
+	interface IBuildOutputContextProvider
+	{
+		bool IsShowingDiagnostics { get; }
+		string SearchString { get; }
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/IBuildOutputContextProvider.cs]: this is not used

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs [hunk 4]
```diff
@@ -343,16 +343,24 @@ public static BuildOutputNode SearchFirstNode (this BuildOutputNode buildOutputN
...
-				return GettextCatalog.GetString ("{0}:{1:d2} min", duration.Minutes, duration.Seconds);
+			if (includeDiagnostics) {
+				if (duration.TotalHours >= 1) {
+					return String.Format ("{0,12}", duration.ToString (@"hh\:mm\:ss\.fff"));
+				} else {
+					return String.Format ("{0,12}", duration.ToString (@"mm\:ss\.fff"));
+				}
+			} else {
+				if (duration.TotalHours >= 1) {
+					return String.Format ("{0,7}", GettextCatalog.GetString ("{0}h {1}m", duration.Hours, duration.Minutes));
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.BuildOutputView/BuildOutputNode.cs]: What type is `duration`? If it's a valuetype, please do ToString before passing to methods which take `object` as the parameter

**Pull request merged**